.github/workflows/cd.yml

```diff
@@ -9,35 +8,63 @@ on:
   release:
     types:
       - published
-
 concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
+  group: "${{ github.workflow }}-${{ github.ref }}"
   cancel-in-progress: true
-
 env:
   FORCE_COLOR: 3
-
 jobs:
   dist:
     name: Distribution build
     runs-on: ubuntu-latest
-
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install dependencies
+        run: |
+          pip install requests==2.31.0 pandas==2.1.1 \
+          google-cloud-bigquery==3.12.0 pyarrow==13.0.0 \
+          db-dtypes==1.1.1 pygithub==2.1.1
+        shell: bash
+
+      - name: Authorize Google Cloud
+        uses: google-github-actions/auth@v1
+        with:
+          credentials_json: "${{ secrets.SERVICE_ACCOUNT_KEY }}"
+          create_credentials_file: true
+          export_environment_variables: true
+
+      - name: Run script to get the latest idc index
+        run: |
+          python scripts/python/idc_index_builder.py
+        shell: bash
+        env:
+          GCP_PROJECT_ID: "${{ secrets.GCP_PROJECT_ID }}"
```

Comment:
Instead, consider adding steps like the following in a workflow called `bump-idc-index-version.yml`:

```
      - name: Authorize Google Cloud
        uses: google-github-actions/auth@v1
        with:
          credentials_json: "${{ secrets.SERVICE_ACCOUNT_KEY }}"
          create_credentials_file: true
          export_environment_variables: true

      - name: Set up Python
        uses: actions/setup-python@v5
        with:
          python-version: 3.11

      - name: Retrieve latest idc version
        id: latest-idc-version
        env:
          GCP_PROJECT_ID: "${{ secrets.GCP_PROJECT_ID }}"
        run: |
          result=$(python scripts/python/idc_index_data_manager.py --retrieve-latest-version)
          echo "result [$result]"
          echo "result=$result" >> $GITHUB_OUTPUT

      - name: Extract current version from CMakeLists.txt
        id: cmakelists-idc-version
        result=$(cat CMakeLists.txt| sed -n 's/^set(IDC_INDEX_DATA_VERSION "\([^"]*\)")/\1/p')
        echo "result [$result]"
        echo "result=$result" >> $GITHUB_OUTPUT

      - name: Update CMakeLists.txt
        if: ${{ steps.latest-idc-version.outputs.value != steps.cmakelists-idc-version.outputs.value}}
        # TODO: Add one-liner updating CMakeLists.txt

      - name: Create Pull Request
        if: ${{ steps.latest-idc-version.outputs.value != steps.cmakelists-idc-version.outputs.value}}
        # TODO: Use peter-evans/create-pull-request

     ....
```

Reply:
And in the `CMakeLists.txt`, you would invoke the script for generating the `.csv.zip` and `.parquet` based on the version.

This approach would then ensure you have a source distribution able to generate a wheel outside of the GitHub CI (assuming the GCP_PROJECT_ID is provided)

Reply:
How about the authentication, when users want to generate a wheel out of GH CI? Are we assuming that the users will gcloud SDK installed and authenticated with necessary bigquery permissions?

@fedorov  what's your take?

Reply:
Also, Andrey opined that at this time, we may only want to update the indices for newer IDC versions with a manual trigger. So I assumed that by only pushing a tag the gha will create the indices using the latest idc version and publish to pypi.

Reply:
> indices for newer IDC versions with a manual trigger.

That makes sense, then simply running a "bump script" locally and creating the PR, and pushing a tag afterward will work well.

> How about the authentication, when users want to generate a wheel out of GH CI? Are we assuming that the users will have gcloud SDK installed and authenticated with necessary bigquery permissions?

The purpose of the wheel is to distribute the index data, if someone want to collect new data from bigquery data, they would indeed need the have the credential. That seems reasonable.

Reply:
> I assumed that by only pushing a tag the gha will create the indices using the latest idc version and publish to pypi.

Yes, that's what I thought we should do. I do not think we should keep IDC data version in CMakeLists.txt. I suggest the steps are:

Manually triggered workflow:
1. Retrieve latest IDC version for `idc_current` dataset
2. Run the queries against versioned dataset
3. save CSV and Parquet results for each query as GHA artifacts

Workflow triggered on release tag creation:
1. Retrieve latest IDC version for `idc_current` dataset
2. Run the queries against versioned dataset
3. attach versioned queries as release attachments
4. publish PyPI package with the attached CSV and/or Parquet results for each query

I would not do any PRs/modifications to the repository as part of the workflows. I do not see a need for that.

Reply:
These requriements may not be relevant anymore but the following are addressed.
> Manually triggered workflow:
    - run each of the queries "as is"
    - save CSV and Parquet results for each query as GHA artifacts
    - Workflow triggered on release tag creation:
    - run each of the queries "as is"
    - publish PyPI package with the attached CSV and/or Parquet results for each query
    This should make the code a lot simpler, reducing maintenance, and I do not think we lose much in terms of functionality.